src/mouse/default_gutter_handler.js

```diff
@@ -105,14 +130,28 @@ function GutterHandler(mouseHandler) {
     function hideTooltip() {
         if (tooltipTimeout)
             tooltipTimeout = clearTimeout(tooltipTimeout);
-        if (tooltipAnnotation) {
+        if (tooltipContent) {
             tooltip.hide();
-            tooltipAnnotation = null;
+            tooltipContent = null;
             editor._signal("hideGutterTooltip", tooltip);
             editor.off("mousewheel", hideTooltip);
         }
     }
 
+    function annotationsToSummaryString(annotations) {
+        var isMoreThanOneAnnotationType = false;
```

Comment:
I think this one is not needed at all if you use [Array.join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join). this function could look like:
```js
function annotationsToSummaryString(annotations) {
    const summary = [];
    const annotationTypes = ['error', 'warning', 'info'];
    for (const annotationType of annotationTypes) {
        if (!annotations[annotationType].length) continue;
        const label = annotations[annotationType].length === 1 ? annotationLabels[annotationType].singular : annotationLabels[annotationType].plural;
        summary.push(`${annotations[annotationType].length} ${label}`);
    }
    return summary.join(", ");
}
```

Reply:
good call